week_1/level2_assignment/random.ts

```diff
@@ -0,0 +1,57 @@
+// 과제 조건
+// 1. Member, Dinner interface 만들고 타입 지정하기
+// 2. organize 내부 로직 채우기
+
+interface Member {
+    name: string;
+    group: string;
+}
+
+interface Dinner {
+    member: Array<Member>;
+    shuffle(array_for_shu: Array<Member>): Array<Member>;
+    organize(array_for_org: Array<Member>): void;
+}
```

Comment:
이 방식도 요구하는 바에 잘 만족합니다! 

**+ 코드 역할 분리하기**
그래서 Interface 별도로 관리하는 디렉토리를 만들고, Member와 Dinner 인터페이스를 따로 작성하는 방법이 있습니다요. 그 후에, 각각의 interface를 export하고 해당 random.ts 에 import하여 사용하는 것이 좀 더 좋을 것 같아요~! 관련해서 읽어보면 좋은 [자료](https://www.typescriptlang.org/ko/docs/handbook/modules.html) 공유해드림니다

**+ 화살표 함수 써보기**
ES6에서 야심차게 내놓은 친구가 화살표함수라고 생각하는데, 한번 써보시는 것도 추천드림니다. 우리가 1차 세미나에서 배운 화살표 함수를 쓰면, 인터페이스 객체 내에서 `member:` 처럼 `shuffle:`, `organize:` 로 좀더 통일된 모습으로 인터페이스를 작성할 수 있슴니다. 저는 이렇게 했는데 사실 정답은 없어요 ㅋ ㅋ Dinner 부분만 봐주시면 됩니다!!
```suggestion
interface Member{
    name: string;
    group: string;
}
interface Dinner{
    member: Member[];
    shuffle: (array: Member[]) => Member[];
    organize: (array: Member[]) => void;
}
```
+ 변수명
(타입스크립트 코딩 가이드라인)[https://happygrammer.tistory.com/156]을 참고해보시면 좋슴니다!
변수명은 보통 언더바보다 카멜 케이스를 사용하면 더 가독성이 좋아요

Reply:
👏🥺👍